### PR TITLE
Full Code Review PR for Hermes Frontend

### PR DESCRIPTION
**Pull request by jchate6**
PR to merge Dev into a designated code-review branch.

### CODE REVIEW

File: public/env-config.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// eslint-disable-next-line no-global-assign
```
Comment (jchate6) [inline public/env-config.js:1]: Add comment to reference in `configmap.yaml`

File: helm-chart/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "hermes-frontend.fullname" . }}
+
+data:
+  env-config.js: |
```
[the remark this answers is not in the document]
Comment (jchate6) [reply helm-chart/templates/configmap.yaml:7]: What is used for npm run serve vs npm build?

File: helm-chart/values.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+securityContext: {}
+
+service:
+  type: ClusterIP
+  port: 8989
+
+ingress:
+  enabled: false
+  annotations: {}
+    # kubernetes.io/ingress.class: nginx
```
Comment (jchate6) [inline helm-chart/values.yaml:34]: Add nginx default values for local development here. (Overwritten in values-prod/dev)

File: src/assets/_custom.scss [hunk 4]
```diff
@@ -0,0 +1,10 @@
+$light: #c9e4f2;
+$white: #fbffff;
+$dark: #03032c;
+$primary: #1b486a;
+$secondary: #00897B;
+$info: #ec9061;
+$success: #388E3C;
+$warning: #F57C00;
+$danger: #BF360C;
+$muted: #FAFAFA;
```
Comment (jchate6) [inline src/assets/_custom.scss:10]: Add new line.

File: src/components/MessageForm.vue [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+.extra-input-col {
+  max-width: 20%;
+}
+.submit-button {
+  color: white;
+}
+.clear-button {
+  float: right;
+}
+</style>
```
Comment (jchate6) [inline src/components/MessageForm.vue:418]: New line

File: src/components/NLEventTable.vue [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+.non-localized-event-table /deep/ td {
+  height: calc(1.5em + 0.75rem + 2px);
+}
+
+.non-localized-event-table /deep/ .data-cell {
+  padding: 0.5rem;
+}
+</style>
```
Comment (jchate6) [inline src/components/NLEventTable.vue:108]: New Line

File: src/components/NavHeader.vue [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+</script>
+
+<style lang="scss" scoped>
+#nav {a.router-link-exact-active {
+  color: white;
+};
+width: 100%;
+}
+
+</style>
```
Comment (jchate6) [inline src/components/NavHeader.vue:53]: New Line

File: src/views/SubmitSpectroscopy.vue [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  name: "SubmitSpectroscopy",
+  components: {
+    MessageForm,
+    "spectroscopy-input-table": SpectroscopyInputTable,
+  },
+}
+
+</script>
+
+<style scoped>
```
Comment (jchate6) [inline src/views/SubmitSpectroscopy.vue:22]: I think all of this styling is defunct, and handled in the components.

File: src/views/SubmitSpectroscopy.vue [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+.extra-input-col {
+  max-width: 20%;
+}
+.submit-button {
+  color: white
+}
+.import-button {
+  color: white
+}
+</style>
```
Comment (jchate6) [inline src/views/SubmitSpectroscopy.vue:50]: New line

File: helm-chart/templates/configmap.yaml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "hermes-frontend.fullname" . }}
+
+data:
```
Comment (cmccully) [inline helm-chart/templates/configmap.yaml:6]: This is how we inject the backend url into the production site after using webpack.

File: helm-chart/templates/deployment.yaml [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+              port: http
+          readinessProbe:
+            httpGet:
+              path: /
+              port: http
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          volumeMounts:
+            - name: app-configuration
+              mountPath: /app/env-config.js
```
Comment (cmccully) [inline helm-chart/templates/deployment.yaml:51]: This is where the env config is mounted.

File: helm-chart/values-dev.yaml [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+    cpu: 10m
+    memory: 256Mi
+  limits:
+    cpu: 1000m
+    memory: 1024Mi
+
+ingress:
+  enabled: true
+  annotations:
+    kubernetes.io/ingress.class: nginx-ingress-private
```
Comment (cmccully) [inline helm-chart/values-dev.yaml:14]: This is LCO specific nomenclature for lco.gtn vs lco.global. This makes sure the load balancer picks up the dns stuff.

File: src/components/CandidatesInputTable.vue [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+        {key: "dec", label: "Dec", type: 'text', editable: true, placeholder: "Dec", class: "dec-column"},
+        {
+          key: "discoveryDate",
+          label: "Discovery Date",
+          type: 'text',
+          editable: true,
+          placeholder: "Discovery Date",
+          class: "discovery-date-column"
+        },
+        {key: 'telescope', label: 'Telescope', type: 'text', editable: true, placeholder: "Telescope", class: "telescope-column"},
```
Comment (cmccully) [inline src/components/CandidatesInputTable.vue:64]: Should we make these vertical aligned to be consistent with below?

**Comment by cmccully** [comment]
I've left a very small number of comments throughout. My only real comment is how do we add testing to all of this?

File: src/components/AdditionalDataTable.vue [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+    <!-- Build Additional Data Table -->
+    <b-editable-table striped bordered class="additional-data-table" v-model="items" :fields="fields" :rowUpdate="rowUpdate">
+      <template #cell(isActive)="data">
+        <span v-if="data.value">Yes</span>
+        <span v-else>No</span>
+      </template>
+      <!-- Delete Row -->
+      <template #cell(delete)="row">
+        <div v-b-tooltip.hover title="Remove Row" @click="removeRow(row)">
+          &#128465;
```
[the remark this answers is not in the document]
Comment (jchate6) [reply src/components/AdditionalDataTable.vue:12]: a little trashcan icon.

**Pull request merged**